Item info page

up2code3 opened the pull request.

royemosby commented on 360_F_248426448_NVKLywWqArG2ADUxDq6QprtIzsF82dMF.jpg:
All assets either go in the /public folder or the src/assets folder. There is already a placeholder image in src/assets that can be used.

royemosby commented on src/pages/itemDetails.jsx:
```diff
@@ -1,11 +1,46 @@
+            <Header pageTitle="Item Details"/>
+        <div class='item_details_container'>
+            
+            <img class='itemImage' src='360_F_248426448_NVKLywWqArG2ADUxDq6QprtIzsF82dMF.jpg'/>
```
@up2code3 swap this out with placeholder.svg - you'll see an example of its usage in ItemCard.jsx.

royemosby commented on src/pages/itemDetails.jsx:
```diff
@@ -1,11 +1,46 @@
-            <p>This page shows the item details</p>
-            <Link to="/">Back to Home Page</Link>
+            <Header pageTitle="Item Details"/>
+        <div class='item_details_container'>
```
Make sure that you're looking at your browser's console while developing because Vite's runtime would have flagged the use of `class` as incorrect. Remember to use `className` instead of `class` and `htmlFor` instead of `for` (inside of input elements). More info can be found in the Standard DOM props section in [React Docs](https://react.dev/reference/react-dom/components/common#common-props)

Please update this and all other usages across this component.

royemosby commented on src/pages/itemDetails.jsx:
```diff
@@ -1,11 +1,46 @@
+            
+            <h4 class='itemName'>Item Name</h4>
+            
+            <div class="description-container"> 
+            <h4 class="description" > Description </h4>
+            <textarea class="DescriptionBox" ></textarea>  {/* <-- text area ? */}
+            </div>
+            
+            <div class='Date-Container'>
+            <h4 class="date" > Date Found </h4>
+            <textarea class="DateBox" placeholder="DD/MM/YYYY"> DD/MM/YYYY </textarea>   {/* <-- text area ? */}
+            </div>
+
+            <div class='Location-Container'>
+            <h4 class="location" > Location </h4>
+            <textarea class="locationBox" placeholder='Location' ></textarea>   {/* <-- text area ? */}
+            </div>
+            
+            <div class='Status-Container'>
+            <h4 class="status" > Status </h4>
+            <div class="lostFound" > Lost </div>   
+            </div>
+
+            <div class='Claim-Button-Container'>
+            <button class='button'> Claim</button>
+            </div>
```
- I think that the Figma design is a bit deceptive here - since there's no editing, there should be no `textarea` elements. I believe that the blocks here represent text placeholders.
- Ensure the appropriate use of heading levels - h1, h2, h3, etc. They are meant to establish a hierarchy inside of the document - here it appears they are being used to achieve the desired size. That is a function of CSS.
    - If I were approaching this, I would place the item name in an h2 (the page title is an h1)and the description in h3
    - The description text should be a paragraph.
    - Date found, location, and status would each be paragraphs and then I would use spans to differentiate the "Date Found" -vs- the actual date 06/04/2024, for example.
    - Use styling to adjust them to the appropriate size.
 - There is no "Back to Home Page" in the Figma design. You can optionally implement it in the "<" icon in the header. You will have to update the Header with new props to do that. Conversely, you can leave the header alone and raise how we're going to use implement interactivity on the "<" icon in Monday's meeting which is fine too.
 - Finally, lets use indenting to nest elements in the code. That will make it easier to read.

up2code3 replied:
I dont quite understand the spans...

<div className='date-Container'>
           <h3 className="date" > Date Found </h3>
           <p className="dateBox"> DD/MM/YYYY</p>  
</div>
                

are you saying change the h3 to <p> and then change the exsisting <p> to a <span> ?

royemosby replied:
>BTW, unescaped html elements will render as html in a markdown renderer (that's why the `<p>` disappeared in your comment after you hit "Comment") but I think I know what you're saying here.

The simplest approach would be to remove the div, wrap the entirety of the text that was in that div with a paragraph tag. Then, use a span to wrap the value text *inside* the paragraph.

A span is a sub-paragraph textual element that can be used to style a portion of text differently than the rest. The MDN description is a bit more nuanced but this is how it's usually used. For example:

```html;
<!- instead of this ->
<div className='date-Container'>
    <h3 className="date" > Date Found </h3>
    <p className="dateBox"> DD/MM/YYYY</p>  
</div>

<!- it can be represented like -->
<p className="dateLine" > Date Found <span className="dateBox"> DD/MM/YYYY</span></p>
```

This saves you a div tag and its associated class + it still allows you the flexibility to create all the styles needed for formatting, color, etc.

up2code3 replied:
I just pushed a commit, for these notes. Honestly at this point I think I could use a huddle or zoom meeting to get a better grasp on things.

royemosby commented on src/styles/itemDetails.css:
```diff
@@ -0,0 +1,136 @@
+  display: flex;
+  flex-direction: column;
+  min-height: 100vh;
+  overflow: hidden;
```
Be careful with this - content can be clipped of the bottom of the page and there will be no scroll bar.

up2code3 replied:
Is there a safer option I shoudl change it to?

royemosby commented on src/styles/itemDetails.css:
```diff
@@ -0,0 +1,136 @@
+color: rgb(102, 102, 102);
+}
+.button{
+background-color: #0C2D57; /* Blue */
```
bg color should be #3B3486 to match Figma design

up2code3 replied:
![Screenshot 2024-04-06 150722](https://github.com/Code-the-Dream-School/ffprac-team7-front/assets/113216091/5bf0c6b7-a1c8-4a4a-95ff-dfddbb852b4f)

I thought this was the blue we were using?

royemosby commented on src/assets/360_F_248426448_NVKLywWqArG2ADUxDq6QprtIzsF82dMF.jpg:
If this asset is going to be used, give it a friendlier name. If not, please remove.

up2code3 replied:
I think I have it proper now.  src={placeHolder} and I deleted the 360_F file.

royemosby commented:
@up2code3 just a heads up - your page's layout is starting to veer off of the Figma design. Make sure that you are using the appropriate margins, padding, and width values.

Some details that will get you on the way:

- `.item_details_container` should have matching left and right margins to give you gutters down the left and right side of the page.
- remove the margins of the sub-elements inside `.description_container`, `.date_Container`, and `.location_Container`. `.item_details_container` will have the margins the page needs.
- the flex-direction set on an element will only affect its children - it does nothing to itself so should not be set on the h3 in the `.description_container`.
- `.date-Container`, `.location-Container`, and `.status-Container` should all use `justify-content: space-between`. Bonus - since they all behave the same way, this is a perfect chance to reduce code use by using a single class instead of three.
- get rid of the width on `.DescriptionBox` so it can take up the space given to it by its parent.
- Watch the padding in the gray paragraphs.
- With the above changes (and a few other similar tweaks you may run into) you will not have a problem with overflow so the`.item_details_container` will not need the `overflow:hidden`

Some other notes:

- ensure you're using consistent capitalization and name casing across all of the classes. There are instances of kebab-case, camelCase, snake_case, and PascalCase, and modified_Snake_Case. In CSS, I recommend sticking with kebab-case or camelCase. Some third-party libraries encourage the use of camelCase but kebab-case will work just fine here. Take a look at what is already in the codebase and stick with what is most prevalent.
- don't forget to merge main into your working branch before review - it's been updated.
- [CSS Tricks](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) has an outstanding guide to flexbox that I have been using for years and I heartily recommend a look.

up2code3 commented:
Hey, now that I have an appearance page for comparison , I am going to put a few more touches on this branch to have my page appearance ( Css sTyle  ) match Mariana's.  ( in addition to addressing the previous notes , which I think are mostly done.. )

up2code3 commented:
I just addressed the comments on 

"@up2code3 just a heads up - your page's layout is starting to veer off of the Figma design. Make sure that you are using the appropriate margins, padding, and width values."

I missed that on my last commit.

2 issues I had...

1) I took the width off of the description box and now it comes up a little short , what technique would I use to make it stretch the width, so it looks like the figma?

2) When I took flex-direction: column off of 'Item name" it lined it on the left side of the page and not centered, so I put it back.

everything else I understood.

I think something thats a bit confusing to me , is this particular figma page is the odd one out , its layout is different, the name of the field is beside it , and not on top of it.

royemosby reviewed (approved):
👍
